app/main.py

```diff
@@ -1,3 +1,56 @@
-def shop_trip():
-    # write your code here
-    pass
+from collections import defaultdict
+
+from app.customer import Customer
+from app.load_jsons import load_complete_json
+from app.shop import Shop, shop_visit_price_calculation, checkout_for_products
+
+
+def shop_trip() -> None:
+    data = load_complete_json("app/config.json")
```

Comment:
I have general questions about the structure of your code:
1. The whole OOP concept in application is about creating classes with methods bound to them. So here you create classes but you don't bound any methods to them. IF you did your code would be much cleaner and well structured.
2. This function is way too long most of the logic can be rewritten as methods that you will call on your classes. Example:
You could put most of this logic as methods in class Customer - methods such as calculate_shop_trip_cost, get_customer_budget, buy_products etc and then in your main function you just call customer.calculate_shop_trip_cost and it will make code so much cleaner.
3. On the same note you can add new classes such as Location and put logic of calcualting distance between two locations there and then reuse it in other code. Or you could define a method in Car to calculate the cost of fuel for the ride.
Think how you can refactor your code so that your final shop_trip() function has about 10 lines of code max - the rest should be handled simply by calling one or two methods on customer instance and the rest should happen under the hood.
Let me know if you have questions

Reply:
the hardest part is to refactor the code after you already wrote it and it's all over the modules, so you need to tear it appart again :D